src/bundle-manifest.ts

```diff
@@ -61,6 +92,22 @@ export function generateBundles(depsIndex: TransitiveDependenciesMap):
 }
 
 /**
+ * Given a collection of bundles and a BundleUrlMapper to generate urls for
+ * them, produce a BundleManifest.
+ */
+export function generateBundleManifest(
```

Comment:
I also have old bias against constructor logic, but I think it makes sense to use constructor here.  Its cleaner.